src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp

```diff
@@ -3301,6 +3301,7 @@ CTranslatorRelcacheToDXL::IsIndexSupported
 	// index expressions and index constraints not supported
 	return gpdb::HeapAttIsNull(tup, Anum_pg_index_indexprs) &&
 		gpdb::HeapAttIsNull(tup, Anum_pg_index_indpred) &&
+		IndexIsValid(index_rel->rd_index) &&
```

Comment:
```suggestion
		index_rel->rd_index->indisready &&
```

I'd strongly recommend we directly use the field than the compatibility macro. This macro is only intended for source compatibility with 9.2. Given that Greenplum never had an release that includes 9.2 but not 9.3, and the fact that this macro is removed in upstream Postgres 12, let's just directly access `indisvalid`.

Reply:
Oh, that's a good catch!  I hadn't realized it was later deleted.  Discussion if anyone else is later interested: https://www.postgresql.org/message-id/flat/d419147c-09d4-6196-5d9d-0234b230880a%402ndquadrant.com